src/Features/Core/Portable/Completion/CompletionListProvider.cs

```diff
@@ -22,5 +22,10 @@ internal abstract class CompletionListProvider
         /// only do minimal textual checks to determine if they should be presented.
         /// </summary>
         public abstract bool IsTriggerCharacter(SourceText text, int characterPosition, OptionSet options);
+
+        public virtual bool IsSnippet
```

Comment:
"IsSnippet" seems really weird to ask a provider.  "Hey provider, are you a snippet?"  "Why yes i am!"   Maybe ProvideSnippets?  or IsSnippetProvider?